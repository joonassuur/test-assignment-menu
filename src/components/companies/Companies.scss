.company-container {
  &:hover {
    background: #F7F7FA;
    cursor: pointer;
  }
  &.active {
    background: rgba(55, 164, 71, 0.1);
  }
}