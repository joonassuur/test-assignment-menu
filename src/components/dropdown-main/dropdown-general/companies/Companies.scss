.company-container {
  .material-icons-outlined {
    font-size: 1em;
  }
  width: 100%;
  display: flex;
  justify-content: space-between;
  &:hover {
    background: #f7f7fa;
    cursor: pointer;
  }
  &:active {
    background: #eeeef2;
  }
  &:focus,
  &.active {
    * {
      color: #157123;
    }
    background: rgba(55, 164, 71, 0.1);
    &:focus,
    &:hover {
      background: rgba(55, 164, 71, 0.2) !important;
    }
  }
}
