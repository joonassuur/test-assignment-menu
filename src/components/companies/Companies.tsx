import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { setActiveCompany, getActiveCompany } from "../../redux/Index";
import { Company } from "../../types/Company";
import "./Companies.scss";

function Companies() {
  const dispatch = useDispatch();
  const activeCompany = useSelector(getActiveCompany);

  const companyList: Company[] = [
    {
      name: "name OÃœ",
      subtitle: "subtitle",
      key: "comp1",
    },
    {
      name: "name2 AS",
      subtitle: "subtitle 2",
      key: "comp2",
    },
    {
      name: "name3 AS",
      subtitle: "subtitle 3",
      key: "comp3",
    },
  ];
  const setCompanyActive = (e: Company) => {
    dispatch(setActiveCompany(e));
  };

  return (
    <div className="companies-container">
      {companyList.map((e) => {
        let isActive = "";
        e.key === activeCompany.key ? (isActive = "active") : (isActive = "");
        return (
          <div
            key={e.key}
            className={`item company-container ${isActive}`}
            onClick={() => setCompanyActive(e)}
          >
            <div className="title">{e.name}</div>
            <div className="subtitle">{e.subtitle}</div>
          </div>
        );
      })}
    </div>
  );
}

export default Companies;
