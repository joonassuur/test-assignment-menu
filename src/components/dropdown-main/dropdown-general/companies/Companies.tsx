import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { setActiveCompany, getActiveCompany, setDropdown } from "../../../../redux/Index";
import { Company } from "../../../../types/Company";
import "./Companies.scss";

function Companies() {
  const dispatch = useDispatch();
  const activeCompany = useSelector(getActiveCompany);

  const companyList: Company[] = [
    {
      name: "Name OÃœ",
      subtitle: "Subtitle 1",
      key: "comp1",
    },
    {
      name: "Name2 AS",
      subtitle: "",
      key: "comp2",
    },
    {
      name: "Name3 AS",
      subtitle: "",
      key: "comp3",
    },
  ];
  const setCompanyActive = (e: Company) => {
    dispatch(setActiveCompany(e));
    // close DD after compeny is selected
    dispatch(setDropdown(false))
  };

  return (
    <div className="companies-container">
      {companyList.map((e) => {
        let isActive = "";
        e.key === activeCompany.key ? (isActive = "active") : (isActive = "");
        return (
          <div
            key={e.key}
            className={`item company-container ${isActive}`}
            onClick={() => setCompanyActive(e)}
          >
            <div className="name">
              <div className="title">{e.name}</div>
              <div className="subtitle">{e.subtitle}</div>
            </div>
            {isActive && <span className="material-icons-outlined">check</span>}
          </div>
        );
      })}
    </div>
  );
}

export default Companies;
