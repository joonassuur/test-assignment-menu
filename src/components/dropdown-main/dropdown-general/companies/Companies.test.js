import React from "react";
import TestRenderer from "react-test-renderer";

import Companies from "./Companies";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../../redux/Index");

describe("Companies component", () => {
  test("Matches the snapshot", () => {
    const companies = TestRenderer.create(<Companies />);
    expect(companies.toJSON()).toMatchSnapshot();
  });
});
