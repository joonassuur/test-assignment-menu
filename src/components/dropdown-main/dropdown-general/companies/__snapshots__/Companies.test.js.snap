// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies component Matches the snapshot 1`] = `
<div
  className="companies-container"
>
  <div
    className="item company-container "
    onClick={[Function]}
  >
    <div
      className="name"
    >
      <div
        className="title"
      >
        Name OÃœ
      </div>
      <div
        className="subtitle"
      >
        Subtitle 1
      </div>
    </div>
    
  </div>
  <div
    className="item company-container "
    onClick={[Function]}
  >
    <div
      className="name"
    >
      <div
        className="title"
      >
        Name2 AS
      </div>
      <div
        className="subtitle"
      >
        
      </div>
    </div>
    
  </div>
  <div
    className="item company-container "
    onClick={[Function]}
  >
    <div
      className="name"
    >
      <div
        className="title"
      >
        Name3 AS
      </div>
      <div
        className="subtitle"
      >
        
      </div>
    </div>
    
  </div>
</div>
`;
